internal/backend/runtime/omni/controllers/omni/config_patch_cleanup.go

```diff
@@ -0,0 +1,234 @@
+	ctrl.initDefaults()
+
+	logger = logger.With(zap.Duration("check_interval", ctrl.CheckInterval), zap.Duration("delete_older_than", ctrl.DeleteOlderThan))
+
+	ticker := ctrl.Clock.NewTicker(ctrl.CheckInterval)
+	defer ticker.Stop()
+
+	initial := true
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+			if !initial {
+				continue
+			}
+		case <-ticker.Chan():
+		}
+
+		if err := ctrl.reconcile(ctx, r, logger); err != nil {
+			return fmt.Errorf("reconciliation failed: %w", err)
+		}
+
+		initial = false
+
+		r.ResetRestartBackoff()
+	}
+}
+
+func (ctrl *ConfigPatchCleanupController) reconcile(ctx context.Context, r controller.ReaderWriter, logger *zap.Logger) error {
+	allConfigPatches, err := safe.ReaderListAll[*omni.ConfigPatch](ctx, r)
+	if err != nil {
+		return err
+	}
+
+	var errs error
+
+	for configPatch := range allConfigPatches.All() {
+		if err = ctrl.processConfigPatch(ctx, r, configPatch, logger); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+// processConfigPatch processes a single the ConfigPatch resource, checking if it is orphaned and cleaning it up if it is.
+func (ctrl *ConfigPatchCleanupController) processConfigPatch(ctx context.Context, r controller.ReaderWriter, configPatch *omni.ConfigPatch, logger *zap.Logger) error {
+	logger = logger.With(zap.String("id", configPatch.Metadata().ID()))
+
+	isOrphan, err := ctrl.isOrphan(ctx, r, configPatch)
+	if err != nil {
+		return err
+	}
+
+	if !isOrphan {
+		return nil
+	}
+
+	logger.Info("destroy orphaned config patch")
+
+	destroyReady, err := r.Teardown(ctx, configPatch.Metadata(), controller.WithOwner(""))
+	if err != nil {
+		if state.IsNotFoundError(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	if !destroyReady {
+		return nil
+	}
+
+	if err = r.Destroy(ctx, configPatch.Metadata(), controller.WithOwner("")); err != nil {
+		if state.IsNotFoundError(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+// initDefaults initializes default values for the controller if they are not set.
+func (ctrl *ConfigPatchCleanupController) initDefaults() {
+	if ctrl.CheckInterval == 0 {
+		ctrl.CheckInterval = time.Hour
+	}
+
+	if ctrl.DeleteOlderThan == 0 { // 30 days
+		ctrl.DeleteOlderThan = time.Hour * 24 * 30
+	}
+
+	if ctrl.Clock == nil {
+		ctrl.Clock = clockwork.NewRealClock()
+	}
+}
+
+// isOrphan checks if the ConfigPatch is orphaned.
+func (ctrl *ConfigPatchCleanupController) isOrphan(ctx context.Context, r controller.Reader, configPatch *omni.ConfigPatch) (bool, error) {
+	if configPatch.Metadata().Owner() != "" {
+		return false, nil
+	}
+
+	if ctrl.Clock.Now().Sub(configPatch.Metadata().Updated()) < ctrl.DeleteOlderThan {
+		return false, nil
+	}
+
+	if _, isSystemPatch := configPatch.Metadata().Labels().Get(omni.LabelSystemPatch); isSystemPatch {
+		return false, nil
+	}
```

Comment:
I think it might make sense to also delete patches that have system patch labels.
User can still create them with the label.